frontend/src/scenes/notebooks/Nodes/NotebookNodeSurvey.tsx

```diff
@@ -0,0 +1,142 @@
+                                <span className="p-2">{survey.description}</span>
+                            </>
+                        )}
+                        {!survey.start_date ? (
+                            <>
+                                <LemonDivider className="my-0" />
+                                <div className="p-2">
+                                    <SurveyReleaseSummary id={id} survey={survey} hasTargetingFlag={hasTargetingFlag} />
+
+                                    <div className="w-full flex flex-col items-center">
+                                        <SurveyAppearance
+                                            type={survey.questions[0].type}
+                                            surveyQuestionItem={survey.questions[0]}
+                                            appearance={survey.appearance || defaultSurveyAppearance}
+                                            question={survey.questions[0].question}
+                                            description={survey.questions[0].description}
+                                            link={
+                                                survey.questions[0].type === SurveyQuestionType.Link
+                                                    ? survey.questions[0].link
+                                                    : undefined
+                                            }
+                                            readOnly={true}
+                                            onAppearanceChange={() => {}}
+                                        />
+                                    </div>
+                                </div>
+                            </>
+                        ) : (
+                            <>
+                                {/* show results when the survey is running */}
+                                <LemonDivider className="my-0" />
+                                <div className="p-2">
+                                    <SurveyResult disableEventsTable />
+                                </div>
+                            </>
+                        )}
+                    </>
+                ) : null}
+
+                <LemonDivider className="my-0" />
+                <div className="p-2 mr-1 flex justify-end gap-2">
+                    {survey.linked_flag && (
+                        <LemonButton
+                            type="secondary"
+                            size="small"
+                            icon={<IconFlag />}
+                            onClick={(e) => {
+                                e.stopPropagation()
+
+                                if (nextNode?.type.name !== NotebookNodeType.FeatureFlag) {
+                                    insertAfter(buildFlagContent((survey.linked_flag as FeatureFlagBasicType).id))
+                                }
+                            }}
+                            disabledReason={
```

Comment:
Maybe instead of disabling the button this could set the flag node to be active?

Reply:
oooh good idea, I like this! Will do in a follow up for everything